tests/integration/medusa_minio_integration_test.go

```diff
@@ -0,0 +1,61 @@
+package integration
+
+import (
+	. "github.com/k8ssandra/k8ssandra/tests/integration/steps"
+
+	. "github.com/onsi/ginkgo"
+)
+
+var (
+	medusaMinioTestTable    = "medusa_test"
+	medusaMinioTestKeyspace = "medusa"
+)
+
+var _ = Describe("Medusa on MinIO", func() {
+	Context("Deploy K8ssandra with Medusa", func() {
+		It("Create a kind cluster and a namespace", func() {
```

Comment:
Maybe we could make the code more versatile to handle different situations. Having tests handle cleanup/creation is very handy to avoid having to wrap executions in further scripts.

I agree that we need to be able to run integration tests on kind, k3d, minikube, etc... but also GKE, EKS, etc... 

We also need to account for limited resources available when running these integration tests, which don't allow us to create several namespaces that would host different k8ssandra clusters. We need to do some cleanup. I've noticed that deleting namespaces could take a while, probably even longer than deleting a cluster with kind and recreating it, so I'm not sure what's the best way to handle this init phase.

Reply:
Deleting namespace (or actually, uninstalling our installation first) is a important test though.

Reply:
>We also need to account for limited resources available when running these integration tests

We indeed do, but I also do not want to require someone to run kind to run the tests. I think it is perfectly reasonable to run against a GKE cluster for example. I would prefer a separate Makefile target for creating the kind cluster. 

>We need to do some cleanup. I've noticed that deleting namespaces could take a while, probably even longer than deleting a cluster with kind and recreating it, so I'm not sure what's the best way to handle this init phase.

Resource cleanup definitely needs to happen in the tests. Note that some resources are cluster scoped so just deleting the namespace for example is not sufficient.